### PR TITLE
backupccl: simplify test removing unnecessary params

### PR DESCRIPTION
**Pull request by andrewbaptist**
Remove the disabling of many of the jobs to simplify TestRestoreErrorPropagates as it was previously attempting to match any key to the range, now it only matches values with RESTORE in them.

Epic: none

Release note: None

### CODE REVIEW

**Review by arulajmani — APPROVED** [review]
:lgtm: modulo comments. Someone more familiar w this test should have a look as well 

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111211#-:-NfCvk1GAVqu3qJp2PFC:b-l7m7p5)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @andrewbaptist, @rhu713, and @stevendanna)

___
*[`-- commits` line 4 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111211#-NfCtDqo36RdVRTYs3oZ):*
nit: "This patch removes..." 

It's generally recommended to only use the imperative form in the PR title, not the body. See https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/73072807/Git+Commit+Messages. 
___
*[`pkg/ccl/backupccl/backup_test.go` line 6155 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111211#-NfCvTr53vIcRxFwfYD8:-NfCvTr53vIcRxFwfYD9:b-muyra0) ([raw file](https://github.com/cockroachdb/cockroach/blob/03e018f84e9c1f3895addbacbee45a20a87f1dd2/pkg/ccl/backupccl/backup_test.go#L6155)):*
> ```Go
> 	defer cancel()
> 
> 	// Fail put requests that have the word "RESTORE" in the value.
> ```

The idea being that the job table write will have the word "RESTORE" in it?

Separately, should this be "Fail conditional put requests"? 
___
*[`pkg/ccl/backupccl/backup_test.go` line 6168 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111211#-NfCuE3J1kpi5a9RjgZX:-NfCuE3J1kpi5a9RjgZY:b-hb3qjr) ([raw file](https://github.com/cockroachdb/cockroach/blob/03e018f84e9c1f3895addbacbee45a20a87f1dd2/pkg/ccl/backupccl/backup_test.go#L6168)):*
> ```Go
> 				Store: &kvserver.StoreTestingKnobs{
> 					TestingRequestFilter: func(ctx context.Context, ba *kvpb.BatchRequest) *kvpb.Error {
> 						if atomic.LoadInt64(&shouldFail) == 0 {
> ```

Should we be using `atomic.Bool` instead?
___
*[`pkg/ccl/backupccl/backup_test.go` line 6171 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111211#-NfCvN2G2RHjfGjJO5Wv:-NfCvN2G2RHjfGjJO5Ww:b-5ok3cv) ([raw file](https://github.com/cockroachdb/cockroach/blob/03e018f84e9c1f3895addbacbee45a20a87f1dd2/pkg/ccl/backupccl/backup_test.go#L6171)):*
> ```Go
> 							return nil
> 						}
> 						// Intercept ConditionalPut requests with RESTORE in the value and,
> ```

We don't need to intercept regular Puts?

File: pkg/ccl/backupccl/backup_test.go
Comment (stevendanna) [inline pkg/ccl/backupccl/backup_test.go:6199]: You mentioned that you found that the shorter intervals slows does the test. That is pretty interesting. I wonder if that is still true with your other changes here.

**Review by stevendanna — APPROVED** [review]
This seems reasonable to me.

The skip under stress race is a bit unfortunate, but is in line with what we've been seeing elsewhere that we currently have a pretty hard time stress-racing any multi-node test.

**Review by andrewbaptist — COMMENTED** [review]
There was a lot of variability in the values, and they took a while to run, so I didn't run enough tests to have lot of confidence. T the "median" of the runs was:
`this branch`: 21.8s
`master`: 28.0s (slightly slower)
`this branch + ShortIntervals()`: 35.3s (much slower)

So it definitely doesn't help things in race builds. 
With short stress runs the values (20 runs of stress)
`this branch`:  Stats over 20 runs: max = 9.0s, min = 4.2s, avg = 6.5s, dev = 1.4s
`master`: Stats over 20 runs: max = 10.5s, min = 8.5s, avg = 9.6s, dev = 0.6s (slightly slower)
`this branch + disable-job-creation`:  Stats over 20 runs: max = 8.5s, min = 3.4s, avg = 7.2s, dev = 1.2s (not significant)
`this branch + ShortIntervals()`: Stats over 20 runs: max = 15.8s, min = 9.9s, avg = 12.9s, dev = 1.9s (much slower)

Given this, I'm going to leave it as it is.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/111211#-:-NfH9iEL2VgQUN9dm3j1:b-5sttdv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @arulajmani, @rhu713, and @stevendanna)

___
*[`-- commits` line 4 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111211#-NfCtDqo36RdVRTYs3oZ):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

nit: "This patch removes..." 

It's generally recommended to only use the imperative form in the PR title, not the body. See https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/73072807/Git+Commit+Messages.
</blockquote></details>

Thanks. Fixed. 
___
*[`pkg/ccl/backupccl/backup_test.go` line 0 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111211#gh-1336894069:-NfHC6Hv-2gnc_zkiR67:bo3qnx7) ([raw file](https://github.com/cockroachdb/cockroach/blob/03e018f84e9c1f3895addbacbee45a20a87f1dd2/pkg/ccl/backupccl/backup_test.go#L0)):*
<details><summary><i>Previously, stevendanna (Steven Danna) wrote…</i></summary><blockquote>

*(Reviewable was [unable](https://github.com/Reviewable/Reviewable/issues/14) to map this GitHub inline comment thread to the right spot — sorry!)*

You mentioned that you found that the shorter intervals slows does the test. That is pretty interesting. I wonder if that is still true with your other changes here.


</blockquote></details>

Yep - see notes above. Maybe we need to kill that setting...
___
*[`pkg/ccl/backupccl/backup_test.go` line 6155 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111211#-NfCvTr53vIcRxFwfYD8:-NfHCBsO3b-jSg6cAMlu:b-vph3s8) ([raw file](https://github.com/cockroachdb/cockroach/blob/03e018f84e9c1f3895addbacbee45a20a87f1dd2/pkg/ccl/backupccl/backup_test.go#L6155)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

The idea being that the job table write will have the word "RESTORE" in it?

Separately, should this be "Fail conditional put requests"?
</blockquote></details>

Correct. I added a little more detail to this comment and fixed to add conditional.
___
*[`pkg/ccl/backupccl/backup_test.go` line 6168 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111211#-NfCuE3J1kpi5a9RjgZX:-NfHCHpWDUAom9t53BoK:bqzusad) ([raw file](https://github.com/cockroachdb/cockroach/blob/03e018f84e9c1f3895addbacbee45a20a87f1dd2/pkg/ccl/backupccl/backup_test.go#L6168)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

Should we be using `atomic.Bool` instead?
</blockquote></details>

Changed. Good idea.
___
*[`pkg/ccl/backupccl/backup_test.go` line 6171 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/111211#-NfCvN2G2RHjfGjJO5Wv:-NfHCJ9bC0CiUeEpCif-:b-po30b8) ([raw file](https://github.com/cockroachdb/cockroach/blob/03e018f84e9c1f3895addbacbee45a20a87f1dd2/pkg/ccl/backupccl/backup_test.go#L6171)):*
<details><summary><i>Previously, arulajmani (Arul Ajmani) wrote…</i></summary><blockquote>

We don't need to intercept regular Puts?
</blockquote></details>

Correct. The requests are always Conditional Puts. I'm not sure exactly why, but this was just from observing the cluster.

**Comment by andrewbaptist** [comment]
bors r=arulajmani,stevendanna 

TFTR!

**Pull request merged**
